lang/cs/Org.Apache.REEF.Network/NetworkService/WritableNetworkService.cs

```diff
@@ -0,0 +1,154 @@
+        /// <param name="messageHandler">The observer to handle incoming messages</param>
+        /// <param name="idFactory">The factory used to create IIdentifiers</param>
+        /// <param name="nameClient">The name client used to register Ids</param>
+        /// <param name="remoteManagerFactory">Writable RemoteManagerFactory to create a 
+        /// Writable RemoteManager</param>
+        [Inject]
+        public WritableNetworkService(
+            [Parameter(typeof (NetworkServiceOptions.NetworkServicePort))] int nsPort,
+            IObserver<WritableNsMessage<T>> messageHandler,
+            IIdentifierFactory idFactory,
+            INameClient nameClient,
+            WritableRemoteManagerFactory remoteManagerFactory)
+        {
+            IPAddress localAddress = NetworkUtils.LocalIPAddress;
+            _remoteManager = remoteManagerFactory.GetInstance<WritableNsMessage<T>>(localAddress, nsPort);
+            _messageHandler = messageHandler;
+
+            NamingClient = nameClient;
+            _connectionMap = new Dictionary<IIdentifier, IConnection<T>>();
+
+            _logger.Log(Level.Info, "Started network service");
+        }
+
+        /// <summary>
+        /// Name client for registering ids
+        /// </summary>
+        public INameClient NamingClient { get; private set; }
+
+        /// <summary>
+        /// Open a new connection to the remote host registered to
+        /// the name service with the given identifier
+        /// </summary>
+        /// <param name="destinationId">The identifier of the remote host</param>
+        /// <returns>The IConnection used for communication</returns>
+        public IConnection<T> NewConnection(IIdentifier destinationId)
+        {
+            if (_localIdentifier == null)
+            {
+                throw new IllegalStateException("Cannot open connection without first registering an ID");
+            }
+
+            IConnection<T> connection;
+            if (_connectionMap.TryGetValue(destinationId, out connection))
+            {
+                return connection;
+            }
+
+            connection = new WritableNsConnection<T>(_localIdentifier, destinationId,
+                NamingClient, _remoteManager, _connectionMap);
+
+            _connectionMap[destinationId] = connection;
+            return connection;
+        }
+
+        /// <summary>
+        /// Register the identifier for the NetworkService with the NameService.
+        /// </summary>
+        /// <param name="id">The identifier to register</param>
+        public void Register(IIdentifier id)
+        {
+            _logger.Log(Level.Info, "Registering id {0} with network service.", id);
+
+            _localIdentifier = id;
+            NamingClient.Register(id.ToString(), _remoteManager.LocalEndpoint);
+
+            // Create and register incoming message handler
+            var anyEndpoint = new IPEndPoint(IPAddress.Any, 0);
```

Comment:
I merged with origin master. I still do not see the above mentioned change there